WooCommerce/WooCommerceTests/ViewRelated/Products/Categories/ProductCategoryListViewModelTests.swift

```diff
@@ -183,6 +183,70 @@ final class ProductCategoryListViewModelTests: XCTestCase {
         // Then
         XCTAssertEqual(viewModel.selectedCategories.count, 0)
     }
+
+    func test_searching_returns_subcategories() {
+        // Given
+        let siteID: Int64 = 132
+        let category1 = Yosemite.ProductCategory(categoryID: 33, siteID: siteID, parentID: 0, name: "Food", slug: "food")
+        insert(category1)
+        let category2 = Yosemite.ProductCategory(categoryID: 12, siteID: siteID, parentID: 33, name: "Oriental", slug: "oriental")
+        insert(category2)
+
+        let viewModel = ProductCategoryListViewModel(siteID: siteID, storageManager: storageManager)
+        XCTAssertEqual(viewModel.categoryViewModels.count, 2) // confidence check
+
+        // When
+        viewModel.searchQuery = "orien"
+
+        // Then
+        waitFor { promise in
+            // wait for 500 milliseconds due to debouncing
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
+                promise(())
+            }
+        }
+        XCTAssertEqual(viewModel.categoryViewModels.count, 1)
+    }
+
+    func test_searchQuery_with_space_returns_results_without_space() {
+        // Given
+        let siteID: Int64 = 132
+        let category1 = Yosemite.ProductCategory(categoryID: 33, siteID: siteID, parentID: 0, name: "Western", slug: "western")
+        insert(category1)
+        let category2 = Yosemite.ProductCategory(categoryID: 12, siteID: siteID, parentID: 0, name: "Oriental", slug: "oriental")
+        insert(category2)
+
+        let viewModel = ProductCategoryListViewModel(siteID: siteID, storageManager: storageManager)
+        XCTAssertEqual(viewModel.categoryViewModels.count, 2) // confidence check
+
+        // When
+        viewModel.searchQuery = " western "
+
+        // Then
+        waitFor { promise in
+            // wait for 500 milliseconds due to debouncing
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
Can we specific debounce time in a private property?

Reply:
Done in c2330856c469e3d010818602efec18a5bbcf6cbb.